autopilot/contracts.go

```diff
@@ -39,3 +77,244 @@ func (ap *Autopilot) defaultContracts() ([]worker.Contract, error) {
 	}
 	return contracts, nil
 }
+
+func (ap *Autopilot) contractLoop() {
+	for {
+		select {
+		// TODO: use build var for loop interval
```

Comment:
maybe we can use a `chan struct{}`, and wire it up to a timer in prod, but trigger it manually in testing

Reply:
not a fan of the `build` package? I always thought that was a very useful feature in `skyd`

Reply:
build tags are kinda cool but they don't play well with tooling (e.g. your IDE can't just run `go test`, it has to run `go test -tags=testing` etc.). I would rather pass any config values explicitly. And just avoid defining separate "testing" values entirely, if possible. The testing environment should match the release environment as closely as possible.

Relatedly, I am *very* wary of any sort of timer-based loop now. I really don't want our test suite to be full of "retry" loops that poll every X milliseconds until some time-based action completes. Basically, core logic should not be tightly coupled to specific intervals of time. Even networking stuff, like dial timeouts and read/write deadlines, should ideally be handled by context cancellation (which can, of course, be triggered by a timer somewhere up the stack).

Reply:
ok, added it as a TODO